### PR TITLE
RD-2475 Add dropdown menu to upload blueprint in blueprints widget.

### PR DESCRIPTION
**Pull request by stepek**
Added dropdown menu to blueprint upload button. In next PR will be connected with the upload from marketplace modal.

### CODE REVIEW

File: app/translations/en.json [hunk 1]
```diff
@@ -655,6 +655,11 @@
                         "placeholder": "Provide the image file URL or click browse to select a file"
                     }
                 }
+            },
+            "blueprintUploadActionsMenu": {
+              "uploadFromMarketplace": "Upload from Marketplace",
+              "uploadFromPackage": "Upload a blueprint package",
+              "generateInComposer":  "Generate in the Composer"
```
Comment (kubama) [inline app/translations/en.json:662]: There is already "blueprintUpload" section above - wouldn't it make sense to name this section just "actionsMenu" and put it there?

File: widgets/common/src/UploadActionsMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    { name: 'generateInComposer', key: 'generateInComposer', content: translate('generateInComposer') }
+];
+
+type ModalName = typeof menuItems[number]['name'];
+
+interface UploadActionsMenuProps {
+    toolbox: Stage.Types.Toolbox;
+}
+
+const UploadActionsMenu: FunctionComponent<UploadActionsMenuProps> = ({ children, toolbox }) => {
```
Comment (kubama) [inline widgets/common/src/UploadActionsMenu.tsx:16]: How about naming this component `BlueprintUploadActionsMenu` (as opposed to plugins upload?)

**Comment by qooban** [comment]
I'll review this PR tomorrow.

**Comment by Gelio** [comment]
I won't review this PR :sweat_smile: I'll unassign myself

File: widgets/common/src/UploadActionsMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import { ComponentProps, FunctionComponent } from 'react';
+
+const translate = Stage.Utils.getT('widgets.common.blueprintUploadActionsMenu');
+const menuItems: ComponentProps<typeof Stage.Basic.Menu.Item>[] = [
+    { name: 'uploadFromMarketplace', key: 'uploadFromMarketplace', content: translate('uploadFromMarketplace') },
+    { name: 'uploadFromPackage', key: 'uploadFromPackage', content: translate('uploadFromPackage') },
+    { name: 'generateInComposer', key: 'generateInComposer', content: translate('generateInComposer') }
+];
+
+type ModalName = typeof menuItems[number]['name'];
```
Comment (qooban) [inline widgets/common/src/UploadActionsMenu.tsx:10]: I'm not sure about the naming here. It's more `ActionName` than `ModalName` as for `generateInComposer` we don't show modal. The other names should be updated accordingly IMHO.

File: widgets/common/src/UploadActionsMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+    } = Stage;
+    const [openedModalName, setOpenedModalName] = React.useState<ModalName | ''>('');
+    const handleMenuClick: ComponentProps<typeof Menu>['onItemClick'] = React.useCallback(
+        (_, { name }) => {
+            setOpenedModalName(name);
+        },
+        [setOpenedModalName]
+    );
+    const hideModal = React.useCallback(() => {
+        setOpenedModalName('');
```
Comment (qooban) [inline widgets/common/src/UploadActionsMenu.tsx:28]: Maybe we should use `null` here instead of empty string?

**Review by qooban — COMMENTED** [review]
Added some comments, mostly related to minor, code style issues - naming, components used, etc.

File: widgets/common/src/UploadActionsMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+                default:
+                    return null;
+            }
+        },
+        [toolbox, hideModal]
+    );
+
+    return (
+        <>
+            <PopupMenu className="deploymentActionsMenu">
```
[the remark this answers is not in the document]
Comment (stepek) [reply widgets/common/src/UploadActionsMenu.tsx:54]: 1. copy&paste
2. Will change

File: widgets/common/src/UploadActionsMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+            const { UploadBlueprintModal } = Stage.Common;
+
+            switch (name) {
+                case 'uploadFromMarketplace':
+                    // TODO RD-2476
+                    return null;
+                case 'uploadFromPackage':
+                    return <UploadBlueprintModal open onHide={hideModal} toolbox={toolbox} />;
+                case 'generateInComposer':
+                    window.open(`/composer/`, '_blank');
```
[the remark this answers is not in the document]
Comment (stepek) [reply widgets/common/src/UploadActionsMenu.tsx:42]: If it's ok to add common function for that i'll change it. Have this in one place can be more useful

File: widgets/common/src/UploadActionsMenu.tsx (hunk 4 above)
Comment (stepek) [reply widgets/common/src/UploadActionsMenu.tsx:28]: i set empty string to prevent type changing of state value.

**Pull request merged**